src/lib.rs

```diff
@@ -98,17 +98,25 @@ pub fn run_book(
   parallel: bool,
   debug: bool,
   linear: bool,
-  skip_warnings: bool,
+  errors: bool,
+  mut warning_opts: WarningOpts,
   opts: Opts,
 ) -> Result<(Term, DefNames, RunInfo), String> {
   let CompileResult { core_book, hvmc_names, labels, warnings } = compile_book(&mut book, opts)?;
+  let warnings = warning_opts.filter(warnings);
 
-  if !warnings.is_empty() {
-    let warnings = warnings.iter().join("\n");
-    if !skip_warnings {
-      return Err(format!("{warnings}\nCould not run the code because of the previous warnings"));
-    } else {
-      eprintln!("{warnings}");
+  // if both enabled, ignore the all option
+  if warning_opts.all && errors {
+    warning_opts.all = false;
+  }
+  if !warning_opts.all {
+    if !warnings.is_empty() {
+      let warnings = warnings.iter().join("\n");
+      if errors {
```

Comment:
We should deny on individual warnings. Just as suggested below a `is_warning_enabled` function, we could have a `is_warning_denied` and filter those that are true.

I also suggest having `-D<warning>` and `-Dno-<warning>` just like for warnings and optimizations.